### PR TITLE
General IDL updates

### PR DESCRIPTION
**Pull request by saschanaz**
General updates that introduce no TS type change.

### CODE REVIEW

File: inputfiles/idl/HTML - Form elements.widl [hunk 1]
```diff
@@ -76,9 +76,10 @@ interface HTMLOptGroupElement : HTMLElement {
 };
 
 [Exposed=Window,
- HTMLConstructor,
- NamedConstructor=Option(optional DOMString text = "", optional DOMString value, optional boolean defaultSelected = false, optional boolean selected = false)]
+ LegacyFactoryFunction=Option(optional DOMString text = "", optional DOMString value, optional boolean defaultSelected = false, optional boolean selected = false)]
```
Comment (sandersn) [inline inputfiles/idl/HTML - Form elements.widl:79]: this change resulted in deleting the `Option` constructor. I think it should remain; do we need to update our code to understand `LegacyFactoryFunction` instead (or in addition)?

File: baselines/dom.generated.d.ts [hunk 2]
```diff
@@ -19408,9 +19408,6 @@ declare var Audio: {
 declare var Image: {
     new(width?: number, height?: number): HTMLImageElement;
 };
-declare var Option: {
-    new(text?: string, value?: string, defaultSelected?: boolean, selected?: boolean): HTMLOptionElement;
-};
```
Comment (orta) [inline baselines/dom.generated.d.ts:19413]: This one feels a bit off to me, this still works in a browser:

```js
new Option()
```

Comment (orta) [reply baselines/dom.generated.d.ts:19413]: Perhaps bad WIDL on the specs?

File: inputfiles/idl/HTML - Form elements.widl (hunk 1 above)
Comment (saschanaz) [reply inputfiles/idl/HTML - Form elements.widl:79]: Oops, yes exactly. I'll revert this change for now and write a new PR for this.

**Pull request merged**